src/Avalonia.Controls/Control.cs

```diff
@@ -329,5 +347,55 @@ protected override void OnKeyUp(KeyEventArgs e)
                 }
             }
         }
+
+        protected override void OnPropertyChanged<T>(AvaloniaPropertyChangedEventArgs<T> change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == FlowDirectionProperty)
+            {
+                InvalidateFlowDirection();
+                
+                foreach (var visual in VisualChildren)
+                {
+                    if (visual is Control child)
+                    {
+                        child.InvalidateFlowDirection();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Computes the <see cref="IVisual.IsMirrorTransform"/> value according to the 
+        /// <see cref="FlowDirection"/> and <see cref="BypassFlowDirectionPolicies"/>
+        /// </summary>
+        public virtual void InvalidateFlowDirection()
```

Comment:
Can be made protected and should only be called by the PropertyChanged handler

Reply:
Again, I need to call him from the parent of his children, and also from `ComboBox`.

Reply:
"protected" still should be available there, isn't it?
Children control inherit Control type, so it should be fine.
But I haven't double checked it and can be wrong.

Reply:
I meant children in the visual tree, not heirs.